crates/uv-distribution-types/src/index_url.rs

```diff
@@ -389,6 +411,20 @@ impl<'a> IndexLocations {
     }
 }
 
+impl From<&IndexLocations> for UrlAuthPolicies {
+    fn from(index_locations: &IndexLocations) -> UrlAuthPolicies {
+        UrlAuthPolicies::from_tuples(index_locations.indexes().map(|index| {
+            let mut url = index
+                .url()
+                .root()
+                .unwrap_or_else(|| index.url().url().clone());
+            url.set_username("").ok();
+            url.set_password(None).ok();
```

Comment:
I wonder if this should be done via `CanonicalUrl`...? That would handle other forms of normalization, but then we'd probably also need to go through `CanonicalUrl` when matching against the policies later on.

Reply:
Just as a note, in the `UrlTrie` we use for credentials, we just iterate on `path_segments`.